### ISSUE TITLE
Tests don't run with setup.py test

### ISSUE THREAD

**Issue by chrismattmann** [chrismattmann/tika-python#51]
Tests seem to only run with:

```python2.7 tika/tests/tests_params.py```

When trying to run through setuptools (note first run python2.7 setup.py develop) I get:

```
test_content (tika.tests.tests_params.RemoteTest) ... ERROR
test_meta (tika.tests.tests_params.RemoteTest) ... ERROR
test_true (tika.tests.tests_params.RemoteTest) ... ERROR

======================================================================
ERROR: test_content (tika.tests.tests_params.RemoteTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mattmann/git/tika-python/tika/tests/tests_params.py", line 48, in setUp
    self.param1 = tika.parser.from_file(self.param1)
  File "/usr/local/xdata-employment/python2.7/lib/python2.7/site-packages/tika/parser.py", line 24, in from_file
    jsonOutput = parse1('all', filename, serverEndpoint)
  File "/usr/local/xdata-employment/python2.7/lib/python2.7/site-packages/tika/tika.py", line 140, in parse1
    path, type = getRemoteFile(urlOrPath, '/tmp')
  File "/usr/local/xdata-employment/python2.7/lib/python2.7/site-packages/tika/tika.py", line 280, in getRemoteFile
    urlp = urlparse(urlOrPath)
  File "/Users/mattmann/git/buildout.python/parts/opt/lib/python2.7/urlparse.py", line 143, in urlparse
    tuple = urlsplit(url, scheme, allow_fragments)
  File "/Users/mattmann/git/buildout.python/parts/opt/lib/python2.7/urlparse.py", line 182, in urlsplit
    i = url.find(':')
AttributeError: 'NoneType' object has no attribute 'find'

======================================================================
ERROR: test_meta (tika.tests.tests_params.RemoteTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mattmann/git/tika-python/tika/tests/tests_params.py", line 48, in setUp
    self.param1 = tika.parser.from_file(self.param1)
  File "/usr/local/xdata-employment/python2.7/lib/python2.7/site-packages/tika/parser.py", line 24, in from_file
    jsonOutput = parse1('all', filename, serverEndpoint)
  File "/usr/local/xdata-employment/python2.7/lib/python2.7/site-packages/tika/tika.py", line 140, in parse1
    path, type = getRemoteFile(urlOrPath, '/tmp')
  File "/usr/local/xdata-employment/python2.7/lib/python2.7/site-packages/tika/tika.py", line 280, in getRemoteFile
    urlp = urlparse(urlOrPath)
  File "/Users/mattmann/git/buildout.python/parts/opt/lib/python2.7/urlparse.py", line 143, in urlparse
    tuple = urlsplit(url, scheme, allow_fragments)
  File "/Users/mattmann/git/buildout.python/parts/opt/lib/python2.7/urlparse.py", line 182, in urlsplit
    i = url.find(':')
AttributeError: 'NoneType' object has no attribute 'find'

======================================================================
ERROR: test_true (tika.tests.tests_params.RemoteTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mattmann/git/tika-python/tika/tests/tests_params.py", line 48, in setUp
    self.param1 = tika.parser.from_file(self.param1)
  File "/usr/local/xdata-employment/python2.7/lib/python2.7/site-packages/tika/parser.py", line 24, in from_file
    jsonOutput = parse1('all', filename, serverEndpoint)
  File "/usr/local/xdata-employment/python2.7/lib/python2.7/site-packages/tika/tika.py", line 140, in parse1
    path, type = getRemoteFile(urlOrPath, '/tmp')
  File "/usr/local/xdata-employment/python2.7/lib/python2.7/site-packages/tika/tika.py", line 280, in getRemoteFile
    urlp = urlparse(urlOrPath)
  File "/Users/mattmann/git/buildout.python/parts/opt/lib/python2.7/urlparse.py", line 143, in urlparse
    tuple = urlsplit(url, scheme, allow_fragments)
  File "/Users/mattmann/git/buildout.python/parts/opt/lib/python2.7/urlparse.py", line 182, in urlsplit
    i = url.find(':')
AttributeError: 'NoneType' object has no attribute 'find'

----------------------------------------------------------------------
Ran 7 tests in 6.722s

FAILED (errors=3)
[chipotle:~/git/tika-python] mattmann% 
```

Any ideas @kdurril and/or @nutjob4life?

**Comment by kdurril** [comment]
My install isn't working after a build, so this I'm not even getting to your position. I may need to ask for additional help.

**Comment by chrismattmann** [comment]
Got it @kdurril ok. Let me know.

**Comment by chrismattmann** [comment]
@nutjob4life any ideas?

**Comment by nutjob4life** [comment]
When you run tests through setup.py, setuptools looks for any classes derived from unittest.TestCase. It will automatically instantiate an object of those classes, then for each member function that starts with "test", it calls setUp, that test function, and tearDown. It requires a no-args __init__ to work.

So, setuptools sees a class "CreateTest", which derives from unittest.TestCase, instantiates it with no-args __init__, but it has no member functions that start with "test" so done.

Then it sees "RemoteTest". So it instantiates it with no-args __init__, calls setUp(), and bombs because the self.param1 was defaulted to None.

To work around this, define a function called test_suite() which takes no-args and returns a unittest.TestSuite populated with just the tests you actually want run.

**Comment by chrismattmann** [comment]
thanks @nutjob4life this HELPS! *hugs*

**Comment by chrismattmann** [comment]
this closes #51
